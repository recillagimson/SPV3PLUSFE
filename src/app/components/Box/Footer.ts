/** Box footer wrapper */
import styled from 'styled-components/macro';
import { StyleConstants } from 'styles/StyleConstants';

const FooterAction = styled.div<{ border?: boolean; align?: string }>`
  align-self: flex-end;
  padding: 7px 10px 15px;
  text-align: ${p => (p.align ? p.align : 'left')};
  border-top: ${p =>
    p.border ? `1px solid ${StyleConstants.LIGHT_BORDER_COLOR}` : '0'};

  & > * {
    margin: 0 5px;
  }
`;

export default FooterAction;
