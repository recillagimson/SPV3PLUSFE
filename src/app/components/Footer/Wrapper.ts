import styled from 'styled-components/macro';
import { StyleConstants } from 'styles/StyleConstants';

const Wrapper = styled.footer`
  text-align: center;
  padding: 20px;
  font-size: 0.9em;

  p {
    margin: 0 0;
    color: #a9b1b8;
  }

  .authenticated & {
    margin-top: auto;
    border-top: 1px solid ${StyleConstants.BORDER_COLOR};
  }
`;

export default Wrapper;
