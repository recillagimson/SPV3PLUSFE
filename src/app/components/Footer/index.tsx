/**
 * Footer Component
 */
import * as React from 'react';

import Wrapper from './Wrapper';

export default function FooterComponent() {
  return (
    <Wrapper className="site-footer">
      <p>&copy; SquidPay {new Date().getFullYear()}</p>
    </Wrapper>
  );
}
