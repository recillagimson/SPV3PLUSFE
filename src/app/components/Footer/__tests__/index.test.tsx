import React from 'react';
import Footer from '..';
import { render } from '@testing-library/react';

describe('<Footer />', () => {
  it('renders correctly', () => {
    const tree = render(<Footer />).baseElement;
    expect(tree).toMatchSnapshot();
  });
});
