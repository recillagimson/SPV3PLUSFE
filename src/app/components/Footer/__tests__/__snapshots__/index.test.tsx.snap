// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders correctly 1`] = `
.c1 {
  text-align: center;
  padding: 20px;
  font-size: 0.9em;
}

.c1 p {
  margin: 0 0;
  color: #a9b1b8;
}

.c1 p .small {
  display: inline-block;
  font-size: 0.8em;
  opacity: 0.75;
  padding-left: 5px;
}

.authenticated .c0 {
  margin-top: auto;
  border-top: 1px solid #CED4DA;
}

<body>
  <div>
    <footer
      class="c0 c1 site-footer"
    >
      <p>
        Â© 
        2021
         SquidPay Technology Inc.
        <span
          class="small"
        >
          (v
          1.2.0
           build: 
          1077
          )
        </span>
      </p>
    </footer>
  </div>
</body>
`;
